packages/framework/src/Model/Product/ProductDataFactory.php

```diff
@@ -116,6 +125,23 @@ public function __construct(
         $this->pluginDataFormExtensionFacade = $pluginDataFormExtensionFacade;
         $this->productParameterValueDataFactory = $productParameterValueDataFactory;
         $this->pricingGroupFacade = $pricingGroupFacade;
+        $this->availabilityFacade = $availabilityFacade;
+    }
+
+    /**
+     * @required
+     * @internal This function will be replaced by constructor injection in next major
+     * @param \Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityFacade $availabilityFacade
+     */
+    public function setAvailabilityFacade(AvailabilityFacade $availabilityFacade)
+    {
+        if ($this->availabilityFacade !== null && $this->availabilityFacade !== $availabilityFacade) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+        if ($this->availabilityFacade === null) {
+            @trigger_error(sprintf('The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
I don't think so. I can change the message to `in version 10`, but I don't think it's necessary.

But that brings me to another idea. It would be impossible to update from 7.3.5 to 8.1 version...

Reply:
Wow, that will be hardcore

Reply:
I agree. We should discuss it further on the team meeting.